src/main/java/by/exadel/internship/service/impl/SchedulingServiceImpl.java

```diff
@@ -0,0 +1,160 @@
+    }
+
+    private List<UserDTO> checkInterviewOver(FormFullDTO formFullDTO) {
+        FormStatus status = formFullDTO.getFormStatus();
+        if (status.equals(FormStatus.REGISTERED)) {
+            return findAdminForInterview(status);
+        }
+        if (status.equals(FormStatus.ADMIN_INTERVIEW_PASSED)) {
+            return findAdminForInterview(status);
+        }
+        throw new InappropriateRoleException("Form with uuid = " + formFullDTO.getId()
+                + " doesn't need to get users time because Form has "
+                + formFullDTO.getFormStatus() + " status");
+    }
+
+    private List<UserDTO> findAdminForInterview(FormStatus formStatus) {
+        log.info("Finding a user with the desired status");
+        List<UserDTO> userDTOList = userService.getAllByUserRole(STATUS_USER_ROLE_MAP.get(formStatus));
+        userDTOList = userDTOList.stream()
+                .filter(userDTO -> userDTO.getUserTimeSlots().size() != 0)
+                .collect(Collectors.toList());
+        log.info("Return user list with desired status");
+        return userDTOList;
+    }
+
+
+    @Override
+    public void saveUserTime(UserDTO userDTO) {
+        MDCLog.putClassNameInMDC(SIMPLE_CLASS_NAME);
+        log.info("Try to save users free time in DB");
+        userTimeSlotService.saveUserTime(userDTO);
+        log.info("Successfully saved time in DB");
+    }
+
+    @Override
+    public void saveInterviewForForm(UUID formId, UserTimeSlotWithUserIdDTO userDataTime) {
+        MDCLog.putClassNameInMDC(SIMPLE_CLASS_NAME);
+        log.info("Try to save Form with Interview");
+        FormFullDTO formFullDTO = formService.getById(formId);
+        UserDTO userDTO = userService.getById(userDataTime.getUserId());
+        if (formFullDTO.getInterview() != null &&
+                formFullDTO.getFormStatus().equals(FormStatus.ADMIN_INTERVIEW_PASSED) &&
+                userDTO.getUserRole().equals(UserRole.TECH_EXPERT)) {
+            InterviewDTO interviewDTO = formFullDTO.getInterview();
+
+            interviewDTO.setTechSpecialist(userDataTime.getUserId());
+            interviewDTO.setTechInterviewDate(userDataTime.getStartDate());
+
+            formFullDTO.setFormStatus(FormStatus.TECH_INTERVIEW_ASSIGNED);
+            userTimeSlotService.deletedById(userDataTime.getId());
+            formService.updateForm(formFullDTO);
+            log.info("Save HR interview date");
+            return;
+        }
+        if (formFullDTO.getFormStatus().equals(FormStatus.REGISTERED) &&
+                userDTO.getUserRole().equals(UserRole.ADMIN)) {
+            InterviewDTO interviewDTO = new InterviewDTO();
+
+            interviewDTO.setAdmin(userDataTime.getUserId());
+            interviewDTO.setAdminInterviewDate(userDataTime.getStartDate());
+
+            formFullDTO.setInterview(interviewDTO);
+            formFullDTO.setFormStatus(FormStatus.ADMIN_INTERVIEW_ASSIGNED);
+            deleteTime(userDataTime.getId());
```

Comment:
follow the same style. above you call direct this service.

Reply:
Done